CompletableFuture Failed to deserialize response

<!-- If you need to report a security issue please visit https://github.com/apache/dubbo/security/policy -->

- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.0.5
* Operating System version: windows 11
* Java version: 1.8

### Steps to reproduce this issue
code repositorie：https://github.com/lmm1990/dubbo-completableFuture-bug
1. http://localhost:20003/demo/hello?name=mark 
success
2. http://localhost:20003/demo/hello_async?name=mark   
Failed to deserialize response


### Expected Behavior

http://localhost:20003/demo/hello_async?name=mark    
need success

### Actual Behavior

Failed to deserialize response

If there is an exception, please attach the exception trace:

```
java.util.concurrent.CompletionException: org.apache.dubbo.remoting.RemotingException: Failed to deserialize response
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:769)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at org.apache.dubbo.rpc.protocol.dubbo.FutureAdapter.lambda$new$0(FutureAdapter.java:44)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at org.apache.dubbo.remoting.exchange.support.DefaultFuture2.doReceived(DefaultFuture2.java:191)
	at org.apache.dubbo.remoting.exchange.support.DefaultFuture2.received(DefaultFuture2.java:139)
	at org.apache.dubbo.remoting.exchange.support.DefaultFuture2.received(DefaultFuture2.java:128)
	at org.apache.dubbo.rpc.protocol.tri.UnaryClientStream$ClientUnaryInboundTransportObserver.onError(UnaryClientStream.java:116)
	at org.apache.dubbo.rpc.protocol.tri.UnaryClientStream$ClientUnaryInboundTransportObserver.lambda$onComplete$0(UnaryClientStream.java:95)
	at org.apache.dubbo.common.threadpool.serial.SerializingExecutor.run(SerializingExecutor.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.dubbo.common.threadlocal.InternalRunnable.run(InternalRunnable.java:41)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.dubbo.remoting.RemotingException: Failed to deserialize response
	... 10 more
```

--- a/dubbo-demo/dubbo-demo-triple/src/main/java/org/apache/dubbo/demo/GreeterService.java
+++ b/dubbo-demo/dubbo-demo-triple/src/main/java/org/apache/dubbo/demo/GreeterService.java
@@ -19,10 +19,16 @@ package org.apache.dubbo.demo;
 import org.apache.dubbo.demo.hello.HelloReply;
 import org.apache.dubbo.demo.hello.HelloRequest;
 
+import java.util.concurrent.CompletableFuture;
+
 public interface GreeterService {
 
     /**
      * Sends a greeting
      */
     HelloReply sayHello(HelloRequest request);
+
+
+    CompletableFuture<String> sayHelloAsync(String request);
+
 }

--- a/dubbo-demo/dubbo-demo-triple/src/main/java/org/apache/dubbo/demo/GreeterServiceImpl.java
+++ b/dubbo-demo/dubbo-demo-triple/src/main/java/org/apache/dubbo/demo/GreeterServiceImpl.java
@@ -19,6 +19,8 @@ package org.apache.dubbo.demo;
 import org.apache.dubbo.demo.hello.HelloReply;
 import org.apache.dubbo.demo.hello.HelloRequest;
 
+import java.util.concurrent.CompletableFuture;
+
 public class GreeterServiceImpl implements GreeterService {
 
     @Override
@@ -26,5 +28,10 @@ public class GreeterServiceImpl implements GreeterService {
         return HelloReply.newBuilder()
                 .setMessage("Hello " + request.getName())
                 .build();
+    }
+
+    @Override
+    public CompletableFuture<String> sayHelloAsync(String name) {
+        return CompletableFuture.supplyAsync(() -> name);
     }
 }

--- a/dubbo-demo/dubbo-demo-triple/src/main/java/org/apache/dubbo/demo/consumer/ApiConsumer.java
+++ b/dubbo-demo/dubbo-demo-triple/src/main/java/org/apache/dubbo/demo/consumer/ApiConsumer.java
@@ -27,6 +27,7 @@ import org.apache.dubbo.demo.hello.HelloReply;
 import org.apache.dubbo.demo.hello.HelloRequest;
 
 import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeUnit;
 
 public class ApiConsumer {
@@ -53,6 +54,9 @@ public class ApiConsumer {
                 .build());
             TimeUnit.SECONDS.sleep(1);
             System.out.println("Reply: " + reply.getMessage());
+
+            CompletableFuture<String> sayHelloAsync = greeterService.sayHelloAsync("triple");
+            System.out.println("Async Reply: "+sayHelloAsync.get());
         } catch (Throwable t) {
             t.printStackTrace();
         }

--- a/dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/DefaultMultipleSerialization.java
+++ b/dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/DefaultMultipleSerialization.java
@@ -39,9 +39,8 @@ public class DefaultMultipleSerialization implements MultipleSerialization {
     public Object deserialize(URL url, String serializeType, String clz, InputStream os) throws IOException, ClassNotFoundException {
         serializeType = convertHessian(serializeType);
         final Serialization serialization = url.getOrDefaultFrameworkModel().getExtensionLoader(Serialization.class).getExtension(serializeType);
-        final Class<?> aClass = ClassUtils.forName(clz);
         final ObjectInput in = serialization.deserialize(null, os);
-        return in.readObject(aClass);
+        return in.readObject();
     }
 
     private String convertHessian(String ser) {

--- a/dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/DefaultMultipleSerialization.java
+++ b/dubbo-serialization/dubbo-serialization-api/src/main/java/org/apache/dubbo/common/serialize/DefaultMultipleSerialization.java
@@ -18,7 +18,6 @@
 package org.apache.dubbo.common.serialize;
 
 import org.apache.dubbo.common.URL;
-import org.apache.dubbo.common.utils.ClassUtils;
 
 import java.io.IOException;
 import java.io.InputStream;

